traefik.go

```diff
@@ -79,17 +86,25 @@ func main() {
 
 	// listen new configurations from providers
 	go func() {
+
 		for {
-			configuration := <-configurationChan
-			log.Infof("Configuration receveived %+v", configuration)
-			if configuration == nil {
+			configMsg := <-configurationChan
+			log.Infof("Configuration receveived from provider %v: %+v", configMsg.providerName, configMsg.configuration)
+			if configMsg.configuration == nil {
 				log.Info("Skipping empty configuration")
-			} else if reflect.DeepEqual(currentConfiguration, configuration) {
+			} else if reflect.DeepEqual(currentConfigurations[configMsg.providerName], configMsg.configuration) {
 				log.Info("Skipping same configuration")
 			} else {
-				newConfigurationRouter, err := LoadConfig(configuration, globalConfiguration)
+				// Copy configurations to new map so we don't change current if LoadConfig fails
+				newConfigurations := make(configs)
+				for k, v := range currentConfigurations {
+					newConfigurations[k] = v
+				}
+				newConfigurations[configMsg.providerName] = configMsg.configuration
+
+				newConfigurationRouter, err := LoadConfig(newConfigurations, globalConfiguration)
 				if err == nil {
-					currentConfiguration = configuration
```

Comment:
This is so we don't try to enable a config that is invalid.  That is, if someone sends in a config that fails to 'load', it will get ignored and not activated.  We first copy the currentConfigurations to a new map, add the incoming config to the new map and try to LoadConfig it.  If that fails, we don't save the incoming config change and all is still good in the world.